editor/components/font-sizes/with-font-sizes.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import { find, get, reduce, upperFirst } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent, Component, compose } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { getFontSize, getFontSizeClass } from './utils';
+
+const DEFAULT_FONT_SIZES = [];
+
+/**
+ * Higher-order component, which handles font size logic for class generation,
+ * font size value retrieval, and font size change handling.
+ *
+ * @param {...(object|string)} args The arguments should all be strings
+ *                                  Each string contains the font size attribute name e.g: 'fontSize'.
+ *
+ * @return {Function} Higher-order component.
+ */
+export default ( ...fontSizeNames ) => {
+	return createHigherOrderComponent(
+		compose( [
+			withSelect( ( select ) => {
+				const settings = select( 'core/editor' ).getEditorSettings();
+				return {
+					fontSizes: get( settings, [ 'fontSizes' ], DEFAULT_FONT_SIZES ),
```

Comment:
We have `fontSizes` provided in settings. Do we really need this default value?

https://github.com/WordPress/gutenberg/pull/6628/files#diff-6e3d39c654369a46e3c418427b3dbddbR9 - here we don't provide the default value. We should pick one approach.